### ISSUE TITLE
`System.import()` wraps exceptions

### ISSUE THREAD

**Issue by DanielSWolf** [systemjs/systemjs#1309]
When a module loaded by `System.import()` throws an exception while loading, the promise returned by `System.import()` is rejected (as expected). However, the reject value is not the original exception, but a modified copy of it. This makes it impossible to correctly handle exceptions that occur at load time. I'd expect to receive the original exception.

The copy is always of type `Error`, even if the original exception had a more specific type. Also, it mangles the original message by prepending its type, adding tabs, and appending the stack trace (see below).

Here is some demo code that illustrates the problem when loading *throw-test.js*.

**throw.js**
```js
const error = new ReferenceError('line 1\nline2');
console.log('Throwing error from throw.js:');
console.log('Name:', JSON.stringify(error.name));
console.log('Message:', JSON.stringify(error.message));
throw error;
```

**throw-test.js**
```js
System.import('./throw').catch(error => {
  console.log('Receiving error while loading throw.js:');
  console.log('Name:', JSON.stringify(error.name));
  console.log('Message:', JSON.stringify(error.message));
});
```

**Output:**

> Throwing error from throw.js:
> Name: "ReferenceError"
> Message: "line 1\nline2"
> Receiving error while loading throw.js:
> Name: "Error"
> Message: "ReferenceError: line 1\n\tline2\n\t    at eval (http://localhost:7357/8463/test/throw.js:1:15)\n\tEvaluating http://localhost:7357/8463/test/throw.js\n\tError loading http://localhost:7357/8463/test/throw.js"

**Comment by guybedford** [comment]
You can use the 'originalErr' accessor on the provided error.

**Comment by DanielSWolf** [comment]
I can't find `originalErr` in the ECMAScript specs. Is there a reason the exception is wrapped and not simply passed on?

**Comment by guybedford** [comment]
The custom errors are in order to provide a custom SystemJS-specific stack
trace containing the list of module loads / executing module stack
associated with the error. Otherwise if we edit the error itself, then we
would lose the original error message / stack.

**Comment by DanielSWolf** [comment]
I see that it makes sense to return a new, SystemJS-specific error. What caught me was the fact that the new error looks *almost*, but not *exactly* like the original error. It doesn't tell you that it is SystemJS-specific; in fact, it took me about an hour to figure that out.

I suggest you make that aspect obvious by using a meaningful *new* error message such as "Error loading SystemJS module. For the original error object, see 'originalErr' property."

**Comment by asapach** [comment]
I wouldn't want to obfuscate the error message, because it will complicate debugging and troubleshooting.

**Comment by guybedford** [comment]
It's a good point, I've included something along these lines in for 0.19.30:

![screen shot 2016-06-09 at 22 59 05](https://cloud.githubusercontent.com/assets/598730/15946192/fc9336ce-2e95-11e6-9855-98f27d1c4fd8.png)

**Comment by guybedford** [comment]
Released in 0.19.30.
